packages/image/README.md

```diff
@@ -0,0 +1,132 @@
+const config = {
+	kit: {
+-		adapter: adapter()
++		adapter: adapter({ images: true })
+	}
+};
+
+export default config;
+```
+
+## Usage
+
+### Dynamic runtime optimization using one of the providers, i.e. an image CDN:
+
+```svelte
+<script>
+	import Image from '@sveltejs/image';
+</script>
+
+<Image src="/path/to/your/image.jpg" width={1200} height={1800} alt="An alt text" />
+```
+
+`width` and `height` should be the intrinsic width/height of the referenced image - i.e. the dimensions of the image before styling. `alt` should describe the image. All are required. The `src` is transformed by calling `getURL` of the `default` provider provided in the `vite.config.js`.
+
+If you want to optimize an image from an external URL, add its domain to the `domains` config of the plugin.
+
+### Static build time optimization:
+
+Static build time optimization uses `vite-imagetools`, which comes as an optional peer dependency, so you first need to install it:
+
+```bash
+npm install --save-dev vite-imagetools
+```
+
+Use in your `.svelte` components:
+
+```svelte
+<script>
+	import { Image } from '@sveltejs/image';
+</script>
+
+<Image src="./path/to/your/image.jpg" alt="An alt text" />
+```
+
+This optimizes the image at build time using `vite-imagetools`. `width` and `height` are optional as they can be inferred from the source image.
+
+Note that the `src` prop must start with a `.`. `@sveltejs/image` includes a preprocessor that will convert it to the following:
+
+```svelte
+<script>
+	import { Image } from '@sveltejs/image';
+	import YourImage from './path/to/your/image.jpg';
+</script>
+
+<Image src={YourImage} alt="An alt text" />
+```
+
+You can also write it yourself like this, if you want to.
+
+Note that the generated code is a `picture` tag (not an `img` tag as in the case of dynamic providers) wrapping one `source` tag per image type.
+
+### Choosing static vs dynamic
+
+Using the static provider generates the images at build time, so build time may take longer the more images you transform.
+
+Using an image CDN provides more flexibility with regards to sizes and you can pass image sources not known at build time, but it comes with potentially a bit of setup overhead (configuring the image CDN) and possibly usage cost.
+
+You can mix and match both solutions in one project.
+
+### `Image` component props
+
+There are a few things you can customize:
+
+- `priority`: set this to `true` for the most important/largest image on the page so it loads faster
+- `sizes`: If your image is less than full width on one or more screen sizes, add this info here. When using dynamic providers the widths can be adjusted accordingly to produce a more optimal `srcset`. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes) for more info on the attribute
+- `style`: to style the image
+- `class`: to style the image. Be aware that you need to pass classes that are global (i.e. wrapped in `:global()` when coming from a `<style>` tag)
```

Comment:
Something like
```html
<img {...getImage({ src, width, height })} .. />
```
feels better to me at first. But it probably does not work with the static import preprocessor then..?

Reply:
(I don't think refactoring from one to the other is something we need to worry about — it's very unlikely that you'd need to do that, because they're really just different problems that happen to both involve an `<img>` as part of the solution)

Reply:
> I don't think refactoring from one to the other is something we need to worry about — it's very unlikely that you'd need to do that, because they're really just different problems that happen to both involve an <img> as part of the solution

I think a common case is that users will start with the build-time solution and then outgrow it and move to a CDN. The build-time solution is much easier to setup - it doesn't require setting up a CDN account, etc. But as sites grow I imagine many people will end up migrating to a CDN

Reply:
Build time to me means things like icons; run time means things like product shots, which you'd never have in your `src` folder in the first place (except maybe for some very early prototyping that you later replace with real data, which isn't the same as 'outgrowing')

Reply:
For a user perspective (indie SaaS & small startups), I'd prefer to stick with build-time generated images as long as practical speed wise--which could be a while for an indie SaaS with 3-4 marketing pages and low-volume blog. 

Being build-time static means 1.) fewer knobs to get off the ground, and 2.) better guarantees that a DDoS will cost nothing on hosts that provide zero-cost static file serving, which wouldn't be the case with a misconfigured cache, etc. 

A couple reasons to stay static as long as practical for some devs.